packages/ra-core/src/auth/Authenticated.spec.tsx

```diff
@@ -32,36 +49,53 @@ describe('<Authenticated>', () => {
             checkError: jest.fn().mockResolvedValue(''),
             getPermissions: jest.fn().mockResolvedValue(''),
         };
-
+        const store = createAdminStore();
+        const dispatch = jest.spyOn(store, 'dispatch');
         const history = createMemoryHistory();
 
-        const { dispatch } = renderWithRedux(
-            <Router history={history}>
-                <AuthContext.Provider value={authProvider}>
-                    <Authenticated>
-                        <Foo />
-                    </Authenticated>
-                </AuthContext.Provider>
-            </Router>
+        const Login = () => {
+            const location = useLocation();
+
+            return (
+                <div aria-label="nextPathname">
+                    {location.state.nextPathname}
+                </div>
+            );
+        };
+        render(
+            <Provider store={store}>
+                <CoreAdminContext
+                    authProvider={authProvider}
+                    dataProvider={() => Promise.resolve()}
+                    history={history}
+                >
+                    <Routes>
```

Comment:
why are you rendering routes here and not in the previous test? Fon consistency, the two tests should have the same children IMO

Reply:
Because in this test, we want to check we actually redirected users to the login page?

Reply:
As discussed, I didn't change the original tests. Testing more things si out of scope for this PR